src/Datadog.Trace.ClrProfiler.Managed.Loader/Startup.cs

```diff
@@ -1,52 +1,186 @@
+    /// Then, this class loads all of the specified assemblies and calls a well known etry point in those assemblies.
+    /// (See <c>TargetLibraryEntrypointXxx</c> constants in this class.)
+    /// If the specified assemblies do not contain such an entry point, they will be loaded, but nothing will be executed.
+    ///
+    /// This class sets up a basic AppDomain.AssemblyResolve handler to look for the assemblies in a framwork-specific
+    /// subdirectory of DD_DOTNET_TRACER_HOME in addition to the normal probing paths.
+    /// If also allows for some SxS loading using costom Assembly Load Context.
+    ///
+    /// If a target assembly needs additional AssemblyResolve to satisfy its dependencies of for any other reasons,
+    /// it should set up its own as the first thing after its entry point is called.
+    ///
+    /// ! Do not make the AppDomain.AssemblyResolve handler in here more complex !
+    /// If anything, it should be simplified and any special logic should be moved into the respective assemblies
+    /// requested for loading.
+    ///
+    /// ! Also, remember that this assembly is shared between the Tracer's profiler component
+    /// and the Profiler's profiler component. DO not put specialized code here !
     /// </summary>
     public partial class Startup
     {
         /// <summary>
-        /// Initializes static members of the <see cref="Startup"/> class.
-        /// This method also attempts to load the Datadog.Trace.ClrProfiler.Managed .NET assembly.
+        /// The constants <c>TargetLibraryEntrypointMethod</c>, <c>...Type</c> and <c>...Namespace</c> specify
+        /// which entrypoint to call in the specified assemblies. The respective assembly is expected to have
+        /// exactly this entry point or nothing will be invoked.
+        /// The method must be static, the return type of the method must be <c>void</c> and it must have no parameters.
+        /// Before doing anything else, the target assemblies must set up AppDomain AssemblyResolve events that
+        /// make sure that their respective dependencies can be loaded.
         /// </summary>
-        static Startup()
+        public const string TargetLibraryEntrypointMethod = "Run";
+
+        /// <summary> <see cref="Startup.TargetLibraryEntrypointMethod" /> </summary>
+        public const string TargetLibraryEntrypointType = "DllMail";
+
+        /// <summary> <see cref="Startup.TargetLibraryEntrypointMethod" /> </summary>
+        public const string TargetLibraryEntrypointNamespace = "Datadog.AutoInstrumentation";
+
+        private const string TargetLibraryEntrypointFullTypeName = TargetLibraryEntrypointNamespace
+                                                                 + "."
+                                                                 + TargetLibraryEntrypointType;
+
+#pragma warning disable SA1308 // Variable names must not be prefixed (if not this, what is the static prefix?)
```

Comment:
I guess it's just `_` for both statics and non statics? I'm not sure though

Reply:
Hmm, ok, strange, but if that's the convention, then sure ))